### PR TITLE
Refactor DoubleInEncodedValue

### PR DESCRIPTION
**Pull request by ksh8281**
I store its tag information in the same location with PointerValues
It can reduce memory access count when finding type

Signed-off-by: Seonghyun Kim <sh8281.kim@samsung.com>

### CODE REVIEW

File: src/runtime/EncodedValue.h [hunk 1]
```diff
@@ -38,18 +38,23 @@ COMPILE_ASSERT(sizeof(EncodedValueData) == 4, "");
 COMPILE_ASSERT(sizeof(EncodedValueData) == 8, "");
 #endif
 
-#define ENCODED_VALUE_DOUBLE_TAG_IN_DATA 1
-
+#pragma pack(push, 1)
+// DoubleInEncodedValue stores its tag in `this + sizeof(size_t)`
+// the location is same with PointerValues
 class DoubleInEncodedValue : public gc {
```
Comment (clover2123) [inline src/runtime/EncodedValue.h:44]: Actually, `DoubleInEncodedValue` has no gc reference in itself.
So, how about overloading `new` operation with `GC_MALLOC_ATOMIC`?

File: src/runtime/EncodedValue.h [hunk 2]
```diff
@@ -59,13 +64,37 @@ class DoubleInEncodedValue : public gc {
 
     double value()
     {
+#ifdef ESCARGOT_32
+        double ret;
+        uint32_t* buf = reinterpret_cast<uint32_t*>(&ret);
+        buf[0] = m_buffer[0];
+        buf[1] = m_buffer[2];
```
Comment (clover2123) [inline src/runtime/EncodedValue.h:71]: For 32bit platform, double value is separately stored in buffer.
Would these two store/load operations slow down the overall EncodedValue perforamance?

File: src/runtime/EncodedValue.h (hunk 1 above)
Comment (ksh8281) [reply src/runtime/EncodedValue.h:44]: It is already allocated by GC_MALLOC_ATOMIC

File: src/runtime/EncodedValue.h (hunk 2 above)
Comment (ksh8281) [reply src/runtime/EncodedValue.h:71]: you are right. dividing saving space causes slow down of copy(little)
but, If save tag in middle of DoubleInEncodedValue,
we can find it is a {double, object, pointer} with one reading from memory.

**Comment by ksh8281** [comment]
this patch increase octane score by 10~15

**Review by bbrto21 — APPROVED** [review]

File: src/runtime/EncodedValue.h (hunk 1 above)
Comment (clover2123) [reply src/runtime/EncodedValue.h:44]: I see

**Review by clover2123 — APPROVED** [review]
LGTM 👍

**Pull request merged**